tests/test_2d_operation_construction.cpp

```diff
@@ -64,48 +68,83 @@ DEBUG_TriMesh test_collapse(const DEBUG_TriMesh& mesh, const Tuple& e, bool shou
 }
 DEBUG_TriMesh test_collapse(const DEBUG_TriMesh& mesh, long edge_index, bool should_succeed)
 {
-    Tuple e = mesh.tuple_from_id(PrimitiveType::Edge, edge_index);
-    REQUIRE(mesh.id(e, PrimitiveType::Edge) == edge_index);
+    Tuple e = mesh.tuple_from_id(PE, edge_index);
+    REQUIRE(mesh.id(e, PE) == edge_index);
     return test_collapse(mesh, e, should_succeed);
 }
 } // namespace
 
-TEST_CASE("trimesh split collapse factories", "[.]")
+TEST_CASE("trimesh_split_collapse_factories", "[operations][2D]")
 {
+    SECTION("split")
     {
         DEBUG_TriMesh m;
         m = single_triangle();
-        auto e = m.tuple_from_id(PrimitiveType::Edge, 0);
+        auto e = m.tuple_from_id(PE, 0);
 
-        REQUIRE(m.id(e, PrimitiveType::Edge) == 0);
-        REQUIRE(m.id(e, PrimitiveType::Vertex) == 0);
-        REQUIRE(m.id(m.switch_tuple(e, PrimitiveType::Vertex), PrimitiveType::Vertex) == 1);
+        REQUIRE(m.id(e, PE) == 0);
+        REQUIRE(m.id(e, PV) == 0);
+        REQUIRE(m.id(m.switch_tuple(e, PV), PV) == 1);
```

Comment:
i'm thinking maybe we should just create inline functions in TriMesh for switch_vertex / switch_edge / switch_face (and same for ID). they're just convenience tools anyway :shrug:

Reply:
I'll do that after finishing this PR

Reply:
done